lib/util/application-logging.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const utils = module.exports
+const MAX_LENGTH = 1021
+const OUTPUT_LENGTH = 1024
+
+/**
+ * Truncates a string to 1021 characters
+ *
+ * @param {string} str string to truncate
+ * @returns {string}
+ */
+utils.truncate = function truncate(str) {
+  if (str.length > OUTPUT_LENGTH) {
+    return str.substring(0, MAX_LENGTH) + '...'
+  }
+
+  return str
+}
+
+/**
+ * Checks if application_logging and one of the features are enabled
+ *
+ * @param {object} config agent config
+ * @returns {boolean}
+ */
+utils.isApplicationLoggingEnabled = function isApplicationLoggingEnabled(config) {
+  return !!(
+    config.application_logging.enabled &&
+    (config.application_logging.metrics.enabled ||
+      config.application_logging.forwarding.enabled ||
+      config.application_logging.local_decorating.enabled)
+  )
+}
+
+/**
+ * Checks if application_logging and application_logging.metrics are both enabled
+ *
+ * @param {object} config agent config
+ * @returns {boolean}
+ */
+utils.isMetricsEnabled = function isMetricsEnabled(config) {
+  return !!(config.application_logging.enabled && config.application_logging.metrics.enabled)
+}
+
+/**
+ * Checks if application_logging and application_logging.local_decorating are both enabled
+ *
+ * @param {object} config agent config
+ * @returns {boolean}
+ */
+utils.isLocalDecoratingEnabled = function isLocalDecoratingEnabled(config) {
+  return !!(
+    config.application_logging.enabled && config.application_logging.local_decorating.enabled
```

Comment:
prettier is usually really good, this line reads really weird

Reply:
I think it's about the same as the other line I complained about above, but I actually find this one easier to read, because it's not also inside a template string.